### PR TITLE
Keep media names, add number if exists

### LINKED ISSUE #1655: Media Upload Random Name
- Laravel Version: 5.4.38
- Voyager Version: 0.11

Hi,
I am developing an website that will need to upload images to the server, and the media, out-of-the-box, is awesome. But one requirement is to keep the names of the images the same, not the 20 chars random.

Can anyone help me?

Thanks in advance.

### PR DESCRIPTION
**Pull request by emptynick**
Uploading a file named `test.txt` into an empty dir:
1. Uploaded as `test.txt`
2. Uploaded as `test_1.txt`
3. Uploaded as `test_2.txt`

Uploading a file named `test_1.txt` into an empty dir:
1. Uploaded as `test_1.txt`
2. Uploaded as `test_2.txt`
3. Uploaded as `test_3.txt`

And so on.
Tested in multiple sub-directories.
Also fixes a problem uploading files with unknown mimetypes, because Laravel generates the extension based on the mimetype, which is undefined in some cases.
Fixes #1655 fixes https://github.com/the-control-group/voyager/issues/2283 closes https://github.com/the-control-group/voyager/pull/3378

### CODE REVIEW

File: src/Helpers/helpers.php [hunk 1]
```diff
@@ -20,3 +20,15 @@ function voyager_asset($path, $secure = null)
         return route('voyager.assets', $path);
     }
 }
+
+if (!function_exists('get_file_name')) {
+    function get_file_name($name)
```
Comment (fletch3555) [inline src/Helpers/helpers.php:25]: Not sure I like this as a helper function... in fact, I think we should move existing helper functions out into service classes or something anyway.  We can do that in another PR

**Review by fletch3555 — APPROVED** [review]

**Comment by maxkopych** [comment]
Do we have this option to keep image names for image field in BREAD?
